src/inferencer/typeEnvironment.ts

```diff
@@ -0,0 +1,337 @@
+export const primitiveMap = new Map()
+primitiveMap.set('-', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'number' },
+    { argumentTypes: ['number'], resultType: 'number' }
+  ],
+  isPolymorphic: false
+})
+primitiveMap.set('*', {
+  types: [{ argumentTypes: ['number', 'number'], resultType: 'number' }],
+  isPolymorphic: false
+})
+primitiveMap.set('/', {
+  types: [{ argumentTypes: ['number', 'number'], resultType: 'number' }],
+  isPolymorphic: false
+})
+primitiveMap.set('%', {
+  types: [{ argumentTypes: ['number', 'number'], resultType: 'number' }],
+  isPolymorphic: false
+})
+primitiveMap.set('&&', {
+  types: [{ argumentTypes: ['boolean', 'any'], resultType: 'any' }],
+  isPolymorphic: false
+})
+primitiveMap.set('||', {
+  types: [{ argumentTypes: ['boolean', 'any'], resultType: 'any' }],
+  isPolymorphic: false
+})
+primitiveMap.set('!', {
+  types: [{ argumentTypes: ['boolean'], resultType: 'boolean' }],
+  isPolymorphic: false
+})
+
+primitiveMap.set('+', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'number' },
+    { argumentTypes: ['string', 'string'], resultType: 'string' }
+  ],
+  isPolymorphic: true
+})
+primitiveMap.set('===', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'boolean' },
+    { argumentTypes: ['string', 'string'], resultType: 'boolean' }
+  ],
+  isPolymorphic: true
+})
+primitiveMap.set('!==', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'boolean' },
+    { argumentTypes: ['string', 'string'], resultType: 'boolean' }
+  ],
+  isPolymorphic: true
+})
+primitiveMap.set('>', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'boolean' },
+    { argumentTypes: ['string', 'string'], resultType: 'boolean' }
+  ],
+  isPolymorphic: true
+})
+primitiveMap.set('>=', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'boolean' },
+    { argumentTypes: ['string', 'string'], resultType: 'boolean' }
+  ],
+  isPolymorphic: true
+})
+primitiveMap.set('<', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'boolean' },
+    { argumentTypes: ['string', 'string'], resultType: 'boolean' }
+  ],
+  isPolymorphic: true
+})
+primitiveMap.set('<=', {
+  types: [
+    { argumentTypes: ['number', 'number'], resultType: 'boolean' },
+    { argumentTypes: ['string', 'string'], resultType: 'boolean' }
+  ],
+  isPolymorphic: true
+})
+
+primitiveMap.set('display', {
+  types: [
+    { argumentTypes: ['number'], resultType: '' },
```

Comment:
Display takes in a variable number of argument types. 
Result type is the type of the first argument. 
Eg: `const x = display(1); x is 1`